.github/workflows/incremental-deploy.yml

```
@@ -22,13 +29,17 @@ jobs:
           fetch-depth: 0
       - name: Authenticate target org
         run: |
-          sfdx org login sfdx-url --set-default --sfdx-url-file <(echo "${{ secrets.SFDX_AUTH_URL_TARGET_ORG }}")
+          sfdx org login sfdx-url --set-default --sfdx-url-file <(echo "${{ inputs.sfdxAuthUrl }}")
       - name: Deploy changes to target org
-        # Note: This requires the merged PR to only have a single commit or merge commit
         run: |
-          sfdx sgd:source:delta --from "HEAD~1" --to "HEAD" --output "." --ignore .forceignore
+          git merge "${{ inputs.baseRef }}"
+          sfdx sgd:source:delta --from "${{ inputs.baseRef }}" --to "${{ inputs.targetRef }}" --output "." --ignore .forceignore
           echo "# package.xml"
           cat package/package.xml; echo ""
           echo "# destructiveChanges.xml"
           cat destructiveChanges/destructiveChanges.xml; echo ""
-          sfdx force:source:deploy --manifest package/package.xml --postdestructivechanges destructiveChanges/destructiveChanges.xml --wait 30 --testlevel RunLocalTests
+          sfdx force:source:deploy \
+            --manifest package/package.xml \
+            --postdestructivechanges destructiveChanges/destructiveChanges.xml \
+            --wait 30 --testlevel RunLocalTests \
```

```suggestion
            --wait 30 --testlevel RunLocalTests \
            --verbose \
```

Otherwise, sometimes issues are not visible in the output